src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -292,5 +309,127 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            if (!TryReadFileAsSpan(statusFilePath, reusableReader, out ReadOnlySpan<char> statusFileContents))
+            {
+                // Between the time that we get an ID and the time that we try to read the associated stat
+                // file(s), the process could be gone.
+                result = default(ParsedStatus);
+                return false;
+            }
+
+            var results = default(ParsedStatus);
+            while (!statusFileContents.IsEmpty)
+            {
+                int startIndex = statusFileContents.IndexOf(':');
+                int endIndex = statusFileContents.IndexOf('\n');
+
+                // Reached end of file
+                if (endIndex == -1)
+                {
+                    break;
+                }
+
+                ReadOnlySpan<char> title = statusFileContents.Slice(0, startIndex);
+                ReadOnlySpan<char> value = default;
+                // endIndex - startIndex - 1 --> To slice till just before '\n'
+                // endIndex - startIndex - 4 --> To slice till just before ' kB\n'
+                switch (title.ToString())
+                {
+#if DEBUG
+                    case "Pid":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 1);
+                        int.TryParse(value, out results.Pid);
+                        break;
+#endif
+                    case "VmHWM":
+                        value = statusFileContents.Slice(startIndex + 1, endIndex - startIndex - 4);
+                        ulong.TryParse(value, out results.VmHWM);
```

Comment:
Since we expect all of these parses to be successful, we should validate that, e.g. create a bool above this switch:
```C#
bool parsedValue = true;
```
then change each of these to store their result into that bool:
```C#
parsedValue = ulong.TryParse(...);
```
and then below the switch add:
```C#
Debug.Assert(parsedValue);
```
or something like that.

Reply:
Done.